drizzlepac/generate_final_product_filenames.py

```diff
@@ -30,177 +187,156 @@ def run_generator(product_category,obs_info):
                                   'multivisit mosaic product': multivisit_mosaic_product_filename_generator}
 
     # Determine which name generator to use based on input product_category
-    for key in category_generator_mapping.keys():
-        if product_category.startswith(key):
-            generator_name = category_generator_mapping[key]
-            category_num = product_category.replace(key+" ","")
+    category_key = ""
+    for ikey in category_generator_mapping.keys():
```

Comment:
```suggestion
    for ikey in category_generator_mapping:
```
Default behaviour for indexing a dict anyway...